### PR TITLE
[SITE-2632] Update check_client_dependencies to check for Relay

### PR DESCRIPTION
**Pull request by EarthlingDavey**
This PR fixes 2 cases where `WP_REDIS_USE_RELAY=true`

- The class is not checked for existence.
- The drop-in will not run if Redis extension is not installed.

The fix:

- Relay\Relay is correctly checked for existence.
- The drop-in still works if the Redis extension is not installed.

### CODE REVIEW

File: object-cache.php [hunk 1]
```diff
@@ -1226,7 +1226,7 @@ protected function _connect_redis() {
 	 *               not with a message describing the issue.
 	 */
 	public function check_client_dependencies() {
-		if ( ! class_exists( 'Redis' ) ) {
+		if ( ! class_exists( WP_REDIS_USE_RELAY ? 'Relay\Relay' : 'Redis' ) ) {
```
Comment (jazzsequence) [inline object-cache.php:1229]:
```suggestion
		// Check for Relay, but default to Redis.
		if ( ! class_exists( WP_REDIS_USE_RELAY ? 'Relay\Relay' : 'Redis' ) ) {
```

**Review by jazzsequence — APPROVED** [review]
The ternary condition will annoy @pwtyler and is somewhat opaque in context. I suggested adding a comment above that line to explain what's happening. I'm okay with this otherwise.

**Review by jazzsequence — CHANGES_REQUESTED** [review]
if `WP_REDIS_USE_RELAY` constant does not exist, PHP errors are thrown. More logic needed to handle if that constant exists before using it in the condition.

File: object-cache.php (hunk 1 above)
Comment (jazzsequence) [inline object-cache.php:1229]:
```suggestion
		$has_relay = defined( 'WP_REDIS_USE_RELAY' ) && class_exists( 'Relay\Relay' );
		if ( ! $has_relay || ! class_exists( 'Redis' ) ) {
```

**Comment by EarthlingDavey** [comment]
Hi @jazzsequence , thanks for catching this and for the suggestion.

I can pick it up next week.

File: object-cache.php (hunk 1 above)
Comment (pwtyler) [reply object-cache.php]: Ah,I applied this change, but in fact we need to check defined('WP_REDIS_USE_RELAY') && WP_REDIS_USE_RELAY && class_exists( 'Relay\Relay' ), in the event the constant is set and false.

**Comment by EarthlingDavey** [comment]
Hey folks, apologies for not thoroughly testing my first PR. Having read all feedback, would the following be ok?

`if ( ! class_exists( defined('WP_REDIS_USE_RELAY') && WP_REDIS_USE_RELAY ? 'Relay\Relay' : 'Redis' ) ) {`

**Pull request closed without merging**